doc/reference/reference_lua/msgpack.rst

```diff
@@ -657,65 +658,100 @@ and :ref:`YAML <yaml-cfg>`.
 
     **Example:**
 
-    ..  code-block:: lua
+    ..  literalinclude:: /code_snippets/test/msgpack/msgpack_object_test.lua
+        :language: lua
+        :lines: 1,29-32
```

Comment:
Looks like you forgot line 5: line 31 refers to `mp_from_string` defined at line 5.

It's difficult to review these examples. IMO better inline them in the text.

Reply:
Thanks, fixed. We decided to use testable examples to make sure we don't miss any breaking changes that might break the code snippets used in docs. Looks like more 'atomic' examples that show only the minimal API usage would be more convenient for review - `msgpack_object_test.lua` actually mixes examples that can be split up.